src/app/AppLayout/AuthModal.tsx

```diff
@@ -52,39 +52,37 @@ export interface AuthModalProps {
 
 export const AuthModal: React.FunctionComponent<AuthModalProps> = (props) => {
   const context = React.useContext(ServiceContext);
+  const [loading, setLoading] = React.useState(false);
   const addSubscription = useSubscriptions();
 
   const onSave = React.useCallback(
-    (username: string, password: string): Promise<void> => {
-      return new Promise((resolve, reject) => {
-        addSubscription(
-          context.target
-            .target()
-            .pipe(
-              first(),
-              filter((target) => target !== NO_TARGET),
-              map((target) => target.connectUrl),
-              mergeMap((connectUrl) => context.jmxCredentials.setCredential(connectUrl, username, password))
-            )
-            .subscribe((result) => {
-              if (result) {
-                props.onSave();
-                resolve();
-              } else {
-                reject();
-              }
-            })
-        );
-      });
+    (username: string, password: string) => {
+      setLoading(true);
+      addSubscription(
+        context.target
+          .target()
+          .pipe(
+            filter((target) => target !== NO_TARGET),
+            first(),
```

Comment:
I changed the order of `first()` after `filter()` to be wait for any valid target (consistent with other views). However, i don't think it really matters since the modal disables any interaction below it.

Just wanna give a head ups :D